govc/object/save.go

```diff
@@ -0,0 +1,303 @@
+		if cmd.one {
+			return nil
+		}
+
+		root = c.ServiceContent.RootFolder
+
+		for _, p := range content[0].PropSet {
+			if c, ok := p.Val.(types.ServiceContent); ok {
+				for _, ref := range mo.References(c) {
+					all := types.NewBool(true)
+					switch ref.Type {
+					case "LicenseManager", "ServiceManager":
+						all = nil
+					}
+					req.SpecSet = append(req.SpecSet, types.PropertyFilterSpec{
+						ObjectSet: []types.ObjectSpec{{
+							Obj: ref,
+						}},
+						PropSet: []types.PropertySpec{{
+							Type:    ref.Type,
+							All:     all,
+							PathSet: nil,
+						}},
+					})
+				}
+				break
+			}
+		}
+	}
+
+	m := view.NewManager(c)
+	v, err := m.CreateContainerView(ctx, root, cmd.kind, cmd.recurse)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		_ = v.Destroy(ctx)
+	}()
+
+	all := types.NewBool(true)
+	req.SpecSet = append(req.SpecSet, types.PropertyFilterSpec{
+		ObjectSet: []types.ObjectSpec{{
+			Obj:  v.Reference(),
+			Skip: types.NewBool(false),
+			SelectSet: []types.BaseSelectionSpec{
+				&types.TraversalSpec{
+					Type: v.Reference().Type,
+					Path: "view",
+					SelectSet: []types.BaseSelectionSpec{
+						&types.SelectionSpec{
+							Name: "computeTraversalSpec",
+						},
+						&types.SelectionSpec{
+							Name: "datastoreTraversalSpec",
+						},
+						&types.SelectionSpec{
+							Name: "entityTraversalSpec",
+						},
+					},
+				},
+				&types.TraversalSpec{
+					SelectionSpec: types.SelectionSpec{
+						Name: "computeTraversalSpec",
+					},
+					Type: "ComputeResource",
+					Path: "environmentBrowser",
+				},
+				&types.TraversalSpec{
+					SelectionSpec: types.SelectionSpec{
+						Name: "datastoreTraversalSpec",
+					},
+					Type: "Datastore",
+					Path: "browser",
+				},
+				&types.TraversalSpec{
+					SelectionSpec: types.SelectionSpec{
+						Name: "entityTraversalSpec",
+					},
+					Type: "ManagedEntity",
+					Path: "recentTask",
+				},
+			},
+		}},
+		PropSet: []types.PropertySpec{
+			{Type: "EnvironmentBrowser", All: all},
+			{Type: "HostDatastoreBrowser", All: all},
+			{Type: "ManagedEntity", All: all},
+			{Type: "Task", All: all},
+		},
+	})
+
+	res, err := methods.RetrievePropertiesEx(ctx, c, &req)
+	if err != nil {
+		return err
+	}
+	if err = cmd.save(res.Returnval.Objects); err != nil {
+		return err
+	}
+
+	token := res.Returnval.Token
+	for token != "" {
+		cres, err := methods.ContinueRetrievePropertiesEx(ctx, c, &types.ContinueRetrievePropertiesEx{
+			This:  req.This,
+			Token: token,
+		})
+		if err != nil {
+			return err
+		}
+		token = cres.Returnval.Token
+		if err = cmd.save(cres.Returnval.Objects); err != nil {
+			return err
+		}
+	}
+
+	var summary []string
+	for k, v := range cmd.summary {
+		if v == 1 && !cmd.verbose {
+			continue
+		}
+		summary = append(summary, fmt.Sprintf("%s: %d", k, v))
+	}
+	sort.Strings(summary)
+
+	s := ", including"
+	if cmd.verbose {
+		s = ""
+	}
+	fmt.Printf("Saved %d total objects%s:\n", cmd.n, s)
```

Comment:
Could be useful to print out the dir name.
```suggestion
	fmt.Printf("Saved %d total objects to %q%s:\n", cmd.n, cmd.dir, s)
```